BASE/src/ServerTelemetryChannel/Implementation/ApplicationFolderProvider.cs

```diff
@@ -56,22 +60,47 @@ public IPlatformFolder GetApplicationFolder()
             var errors = new List<string>(this.environment.Count + 1);
 
             var result = this.CreateAndValidateApplicationFolder(this.customFolderName, createSubFolder: false, errors: errors);
-            
-            if (result == null)
+
+            // In the constructor, for windows environment we set the value of ApplySecurityToDirectory to SetSecurityPermissionsToAdminAndCurrentUserWindows.
```

Comment:
we can do better here by leveraging Environment.OSVersion to detect the OS. THe current code does thing based on exception being thrown or not - as that was the only way in .netstandard1.3 - Can you refactor this to use Environment.OSVersion to make this decision?